scripts/release_info.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Script that simplifies releases by collecting various information.
+Should be run from the oppia root dir.
+"""
+import collections
+import os
+import re
+import subprocess
+import sys
+
+
+class ChangedBranch(object):
+    def __init__(self, new_branch):
+        get_branch_cmd = 'git symbolic-ref -q --short HEAD'.split()
+        self.old_branch = subprocess.check_output(get_branch_cmd).strip()
+        self.new_branch = new_branch
+        self.is_same_branch = self.old_branch == self.new_branch
+
+    def __enter__(self):
+        if not self.is_same_branch:
+            try:
+                subprocess.check_output(["git", "checkout", self.new_branch])
+            except subprocess.CalledProcessError:
+                print ('\nCould not change to %s. This is most probably because'
+                       ' you are in a dirty state' % self.new_branch)
+                sys.exit(1)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if not self.is_same_branch:
+            subprocess.check_output(["git", "checkout", self.old_branch])
+
+
+Log = collections.namedtuple('Log', ['sha1', 'author', 'email', 'message'])
```

Comment:
Hm, in a sense it's a matter of definition -- I understood this to end up as a tuple when I read it, not really a full-fledged object-derived class. I think, insofar as there's a guideline, it's "small one-line top-level defns above big ones"; usually the former are constants/variables. The aim is to make them easier to find; it's a bit hard somehow to find a one-line thing sandwiched in the middle of two big classes/functions.

(To be clear, though, it's not really a big deal either way.)